client/src/app/shared/components/os-sort-filter-bar/os-sort-filter-bar.component.ts

```diff
@@ -0,0 +1,388 @@
+    }
+
+    /**
+     * Handles the sorting menu/bottom sheet (depending on state of mobile/desktop)
+     */
+    public openSortDropDown(): void {
+        if (this.vp.isMobile){
+            const bottomSheetRef = this.bottomSheet.open(OsSortBottomSheetComponent,
+                { data: this.sorting }
+            );
+            bottomSheetRef.afterDismissed().subscribe(result => {
+                if (result) {
+                    // result can be undefined if dismissed without selecting anything
+                    this.sorting = result;
+                    this.filterData();
+                }
+            });
+        }
+    }
+
+    /**
+     * Catches the escape key to exit the search bar on top; for other keys, it updates the search
+     * @param event a Key press on the search bar
+     * @param value current event.target.value
+     */
+    public applySearch(event: KeyboardEvent, value?: string): void {
+        if (event.keyCode === 27 ) {
+            this._filteredDataSource.filter = '';
+            this.isSearchBar = false;
+        } else {
+            this._filteredDataSource.filter = value;
+        }
+            this.filterData();
+    }
+
+    /**
+     * Helper to listen for the emitted updates from the {@link filterMenu}
+     * @param event
+     */
+    public newFilters(event: OsFilter[]): void {
+        this.filters = event;
+        this.saveCurrentSortFilterState();
+        this.filterData();
+    }
+
+    /**
+     * Reacts to a sorting change in the sorting menu.
+     * TODO: duplicate function in sortBottomSheet
+     * @param item
+     */
+    public sortingChange(item: OsSortingItem<V>): void {
+        if (this.sorting.sortProperty === item.property){
+            this.sorting.sortAscending = !this.sorting.sortAscending;
+        } else {
+            this.sorting.sortProperty = item.property;
+            this.sorting.sortAscending = true;
+        }
+        this.saveCurrentSortFilterState();
+        this.filterData();
+        this.menuTrigger.closeMenu();
+    }
+
+    /**
+     * returns an upper case Label from label or property
+     * TODO: duplicate function in sortBottomSheet
+     * @param item
+     */
+    public getLabel(item: OsSortingItem<V>): string {
+        if (item.label) {
+            return item.label;
+        } else {
+            const itemProperty = item.property as string;
+            return itemProperty.charAt(0).toUpperCase() + itemProperty.slice(1);
+        }
+    }
+
+    /**
+     * Helper to get the proper icon for a sorting entry in the menu
+     * (none/ascending/descending)
+     * @param item
+     * TODO: duplicate function in sortBottomSheet
+     */
+    public getIcon(item: OsSortingItem<V>): string {
+        if (item.property !== this.sorting.sortProperty) {
+            return '';
+        }
+        return this.sorting.sortAscending ? 'arrow_downward' : 'arrow_upward';
+    }
+
+    public checkStoredFilterSort(): void {
```

Comment:
candidate for a function inside a filter (core) service.

Also missing documentation